Fundamentals/Code/Source-Control/index-v9.md

```diff
@@ -0,0 +1,174 @@
+
+So if you've made the decision to try to attempt to source/version control your Umbraco implementation work, perhaps setting up a ['Git Repository'](https://en.wikipedia.org/wiki/Git) - then a frequently asked question is:
+
+### What folders and files should I **exclude** from my source control repository?
+
+There are lots of different possible variations within your working environment that will affect the best way to setup version control. It depends on whether you are:
+
+- Working with a team of developers.
+- How your development environment is setup.
+- Source control repository.
+- And also how you intend to build and deploy your solution to your target production environment (build servers, Web Deploy or good old FTP etc).
+
+However, V9 ships with a `.gitignore` file with a custom Umbraco section, which will make git ignore the files for you, if you use git, the Umbraco specific section looks like this:
+
+```
+##
+## Umbraco CMS .NETCore
+##
+
+# Dont commit Umbraco TEMP folder containing Examine Indexes, NuCache etc
+**/umbraco/Data/TEMP/
+
+# Dont commit files that are generated and cached from the default ImageSharp location
+**/umbraco/mediacache/
+
+# Umbraco backoffice language files
+# Nuget package Umbraco.Cms.StaticAssets will copy them in during dotnet build
+# Customize langguage files in /config/lang/{language}.user.xml
+**/umbraco/config/lang/
+
+# JSON Schema file for appsettings
+# This is auto generated from the build
+**/umbraco/config/appsettings-schema.json
+
+# This is the no-nodes, installer & upgrader pages from Umbraco
+# Nuget package Umbraco.Cms.StaticAssets will copy them in during dotnet build
+**/umbraco/UmbracoWebsite/
+**/umbraco/UmbracoInstall/
+**/umbraco/UmbracoBackOffice/
+
+# Comment out the line below if you wish to change or add any new templates to PartialView Macros
+**/umbraco/PartialViewMacros/
+
+# Umbraco Static Assets of Backoffice
+# Nuget package Umbraco.Cms.StaticAssets will copy them in during dotnet build
+**/wwwroot/umbraco/
+```
+
+For most projects this gitignore will be enough, and this documentation can't be an exhaustive list of how to version control Umbraco in all possible scenarios. However, we will go through the different files, and hopefully it will give you an insight into the anatomy of how an Umbraco website hangs together and therefore which parts to include in version control and which parts not to.
+
+![Typical set of Umbraco Project Folders](images/folder-structure-v9.jpg)
+
+#### The Umbraco Folder
+
+The main folder where the Umbraco CMS resides in, is the `/umbraco` one inside your project.
+
+Some of the contents change when you upgrade Umbraco, and will even be deleted and re-added on a clean and rebuild! No part of your implementation or third party packages installed should update these folders.
+
+From the gitignore above we can see that the content that will change when you upgrade Umbraco and be deleted and re-added on clean and rebuild are:
```

Comment:
```suggestion
For most projects, this gitignore will be enough, and this documentation can't be an exhaustive list of how to version control Umbraco in all possible scenarios. However, we will go through the different files, and hopefully, it will give you an insight into the anatomy of how a Umbraco website hangs together and therefore which parts to include in version control and which parts not to.

![Typical set of Umbraco Project Folders](images/folder-structure-v9.jpg)

#### The Umbraco Folder

The main folder where the Umbraco CMS resides in, is the `/umbraco` one inside your project.

Some of the contents change when you upgrade Umbraco, and will even be deleted and re-added on a clean and rebuild! No part of your implementation or third-party packages installed should update these folders.

From the gitignore above we can see that the content that will change when you upgrade Umbraco and be deleted and re-added on clean and rebuild are:
```